src/sentry/static/sentry/app/components/accordion.jsx

```diff
@@ -0,0 +1,84 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import InlineSvg from 'app/components/inlineSvg';
+import space from 'app/styles/space';
+import {Flex} from 'grid-emotion';
+import Badge from 'app/components/badge';
+import Button from 'app/components/buttons/button';
+
+class AccordionButton extends React.Component {
+  static propTypes = {
+    labelClosed: PropTypes.string.isRequired,
+    labelOpen: PropTypes.string,
+    cutoff: PropTypes.number,
+    children: PropTypes.arrayOf(PropTypes.node).isRequired,
+  };
+
+  static defaultProps = {
+    cutoff: 0,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  handleClick = e => {
+    e.preventDefault();
+    this.setState({isOpen: !this.state.isOpen});
```

Comment:
A better way to update state that depends on current state is to use a callback (because `setState` is async):

```
this.setState((state) => ({isOpen: !state.isOpen}));
```

Reply:
so many good tips